### ISSUE TITLE
[Bug]: Yarn SDKs requires ".pnp.cjs" property in package.json's exports.

### ISSUE THREAD

**Issue by CharlesStover** [yarnpkg/berry#3493]
### Self-service

- [ ] I'd be willing to implement a fix

### Describe the bug

A regression was implemented at some point, seemingly to Yarn itself (`.pnp.cjs`?). If you do not have a `".pnp.cjs": "./.pnp.cjs"` in your `package.json`'s `exports` field, then `yarn sdks vscode` will fail with "Error: Missing ".pnp.cjs" export in "${{ package name }}" package".

```
λ yarn sdks vscode
➤ YN0000: ┌ Generating SDKs inside .yarn/sdks
➤ YN0000: │ ✓ Eslint
➤ YN0001: │ Error: Missing ".pnp.cjs" export in "sentry-react" package
    at bail (F:\git\sentry-react\.pnp.cjs:17643:8)
    at resolve (F:\git\sentry-react\.pnp.cjs:17695:7)
    at applyNodeExportsResolution (F:\git\sentry-react\.pnp.cjs:17970:28)
    at resolveUnqualifiedExport (F:\git\sentry-react\.pnp.cjs:18483:35)
    at resolveRequest (F:\git\sentry-react\.pnp.cjs:18549:99)
    at Object.resolveRequest (F:\git\sentry-react\.pnp.cjs:18624:26)
    at Wrapper.writeFile (F:\git\sentry-react\.yarn\cache\@yarnpkg-sdks-npm-2.4.2-rc.6-793a379d9e-091f94afda.zip\node_modules\@yarnpkg\sdks\lib\generateSdk.js:173:72)
    at Wrapper.writeBinary (F:\git\sentry-react\.yarn\cache\@yarnpkg-sdks-npm-2.4.2-rc.6-793a379d9e-091f94afda.zip\node_modules\@yarnpkg\sdks\lib\generateSdk.js:166:20)
    at generateEslintBaseWrapper (F:\git\sentry-react\.yarn\cache\@yarnpkg-sdks-npm-2.4.2-rc.6-793a379d9e-091f94afda.zip\node_modules\@yarnpkg\sdks\lib\sdks\base.js:8:19)
    at async F:\git\sentry-react\.yarn\cache\@yarnpkg-sdks-npm-2.4.2-rc.6-793a379d9e-091f94afda.zip\node_modules\@yarnpkg\sdks\lib\generateSdk.js:225:33
➤ YN0000: └ Completed
```

### To reproduce

[Here is a snapshot of my `package.json`  and Yarn configuration.](https://github.com/CharlesStover/sentry-react/tree/00267104859dcfac0e2099b1a684df600488e83f)

Running `yarn && yarn sdks vscode` should be enough to trigger this error.

### Environment

```shell
λ yarn dlx -q envinfo --preset jest
D:\Temporary Files\xfs-e9e7fae1\dlx-25464\.pnp.cjs:9582
    throw firstError;
    ^

Error: Cannot find module './envinfo'
```
lol
```


### Additional context

_No response_

**Comment by CharlesStover** [comment]
For additional information, this is occurring whenever `exports` is set. In my case, I am setting the reserved Node exports `"require"` and `"import"`. See: [conditional exports](https://nodejs.org/api/packages.html#packages_conditional_exports).

When those are the only set values, PNP crashes. Not only does `yarn sdks *` crash for any SDK, but it also crashes `eslint` with the same error. Hope this helps.

**Comment by paul-soporan** [comment]
Will be fixed by #3495.

> Error: Cannot find module './envinfo'

I can't repro this, if it's still a problem and you can provide a reproduction you should open a different issue for it.
